appserver/jdbc/jdbc-config/src/main/java/org/glassfish/jdbc/config/validators/JdbcConnectionPoolValidator.java

```diff
@@ -65,23 +68,71 @@ public void initialize(final JdbcConnectionPoolConstraint constraint) {
+            case "ENV":
+                //Check environment variables for requested value
+                if (System.getenv(variableReference[1]) != null && !System.getenv(variableReference[1]).isEmpty()) {
+                    return System.getenv(variableReference[1]);
+                }
+                break;
+            case "MPCONFIG":
+                //Check microprofile config for requested value
+                Config config = ConfigProvider.getConfig();
+                if (config.getValue(variableReference[1], String.class) != null && !config.getValue(variableReference[1], String.class).isEmpty()) {
+                    return config.getValue(variableReference[1], String.class);
+                }
+                break;
+            default:
+                //We got a system variable as no split occured
+                if (systemProps.getProperty(variableReference[0]) != null && !systemProps.getProperty(variableReference[0]).isEmpty()) {
+                    return systemProps.getProperty(variableReference[0]);
+                }
+                break;
+        }
+        
+        //If this point is reached, the variable value could not be found
+        return null; 
+    }
+    
     @Override
     public boolean isValid(final ResourcePool pool,
         final ConstraintValidatorContext constraintValidatorContext) {
 
         if(poolFaults == ConnectionPoolErrorMessages.MAX_STEADY_INVALID) {
             if(pool instanceof JdbcConnectionPool) {
+                
                 JdbcConnectionPool jdbcPool = (JdbcConnectionPool) pool;
                 String maxPoolSize = jdbcPool.getMaxPoolSize();
                 String steadyPoolSize = jdbcPool.getSteadyPoolSize();
-                if(steadyPoolSize == null) {
+                
+                int maxPoolSizeValue = 0;
+                int steadyPoolSizeValue = 0;
+                
+                if (steadyPoolSize == null) {
                     steadyPoolSize = Constants.DEFAULT_STEADY_POOL_SIZE;
+                } else if(steadyPoolSize.startsWith("$")) {
+                    steadyPoolSize = getParsedVariable(steadyPoolSize.substring(2, steadyPoolSize.length() - 1));
+                    if(steadyPoolSize == null) return false;
                 }
+                
                 if (maxPoolSize == null) {
                     maxPoolSize = Constants.DEFAULT_MAX_POOL_SIZE;
+                } else if(maxPoolSize.startsWith("$")) {
+                    maxPoolSize = getParsedVariable(maxPoolSize.substring(2, maxPoolSize.length() - 1));
+                    if(maxPoolSize == null) return false;
                 }
-                if (Integer.parseInt(maxPoolSize) <
-                        (Integer.parseInt(steadyPoolSize))) {
+                
+                //By this point it should be the case that the value is always castable
+                //to an integer
+                maxPoolSizeValue = Integer.parseInt(maxPoolSize);
```

Comment:
This will cause `NumberFormatException` if the string does not have the appropriate numeric value.